tools/links.py

```diff
@@ -98,7 +98,7 @@ def retrieve_major_minor(package: str):
         The major and minor versions of the package.
 
     """
-    with open(PYPROJECT_TOML_FILE, "r") as file:
+    with Path.open(PYPROJECT_TOML_FILE, "r") as file:

```

Comment:
This is not the way of using the `Path` class. I think you may be confused because of the way it shows in the official documentation.

You must instantiate the class before and then use the provided methods. Example:

```python
# Both PYPROJECT_TOML and DOCS_DIRECTORY are Path objects

PYPROJECT_TOML.open(...)
DOCS_DIRECTORY.walk(...)
```